src/context/mod.rs

```diff
@@ -651,6 +653,32 @@ impl Context {
         let acl_permission_result: Result<(), &str> = acl_permission_result.into();
         acl_permission_result.map_err(|_e| RedisError::Str("User does not have permissions on key"))
     }
+
+    #[redismodule_api(RedisModule_AddPostNotificationJob)]
```

Comment:
I think this is more or less what the proc macro does. We can polish it a littile more as you suggesting and come up with an easier implementation.

Reply:
@vityafx I took another approach, the original function is turned into a closure so it can access all the inputs and the raw:: API (after been checked). Let me know what you think.